packages/plugins/users-permissions/server/bootstrap/index.js

```diff
@@ -28,7 +28,15 @@ module.exports = async ({ strapi }) => {
   await getService('users-permissions').initialize();
 
   if (!strapi.config.get('plugin.users-permissions.jwtSecret')) {
-    const jwtSecret = uuid();
+    const jwtSecret = crypto.randomBytes(16).toString('base64');
+
+    if (process.env.NODE_ENV === 'production') {
+      throw new Error(
+        `Missing jwtSecret. Please, set jwtSecret for the users-permissions plugin in config/plugins.js or set environment variable JWT_SECRET (ex: ${jwtSecret}).
```

Comment:
I think we have a bit of inconsistency here. 
For API_TOKEN_SALT we deprecated just setting the env var but not for JWT_SECRET. Is there a specific reason I forgot  about 

I think this might create some confusion in the future, wdyt?

Reply:
It's because JWT_SECRET is auto-generated at start-up when in dev mode. Whereas the other are created when the project is created.
However I can change the error message to only mention the config file and not the env var.

Reply:
Hum then I think there are some changes to do with the generation of the env var too. 

It should only happen in dev mode (right now we generate every time it's not prod, I would generate only if it's dev and throw if node_env is not dev)

And Yeah I think we should not mention the JWT_SECRET it self if we want to be as consistent as possible 🤔

Reply:
So you would just change that:
```diff
-    if (process.env.NODE_ENV === 'production') {
+    if (process.env.NODE_ENV !== 'development') {
```
?